miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppScheme.kt

```diff
@@ -35,6 +36,18 @@ internal class MiniAppScheme private constructor(miniAppId: String) {
         }
     }
 
+    fun appendParametersToUrl(url: String, queryParams: String): String {
+        return if (queryParams.isEmpty()) url
+        else "$url${resolveParameters(queryParams)}"
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    fun resolveParameters(queryParams: String): String {
+        val qusMark = "?"
+        return if (queryParams.contains(qusMark) && queryParams.startsWith(qusMark)) queryParams
```

Comment:
I think that our spec defines the param as this format: `param=value&param2=value` so we don't have to check the `?`.

Reply:
Checking qus mark is a safe call here. At-least we need to filter the qus mark here. Additionally, I was thinking if we also need to verify something else if the params are valid or not.

Reply:
I don't think we should do any additional check whether the params are valid - we should just pass them directly to the mini app and then it's their responsibility to parse and verify. The reason is because it's really difficult for us to predict how the mini app will use the params, so we would probably end up breaking something for some mini apps.

Also, I think that passing additional `?` in the params is technicadlly valid. For example, this is valid: `param?=value?`. Something like `??param=value` is also valid, but I can't think of any situation where they would purposely want that 🤔  So this is probably a good idea to ensure just one beginning `?` here.

Reply:
Yeah, I agree to let miniapp itself to handle the parsing and verification. And, in future, we can add more verification inside SDK based on client's requests.